src/rdf4cpp/rdf/datatypes/xsd/time/DateTimeStamp.cpp

```diff
@@ -0,0 +1,44 @@
+#include <rdf4cpp/rdf/datatypes/xsd/time/DateTimeStamp.hpp>
+
+#include <rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp>
+
+namespace rdf4cpp::rdf::datatypes::registry {
+
+template<>
+capabilities::Default<xsd_dateTimeStamp>::cpp_type capabilities::Default<xsd_dateTimeStamp>::from_string(std::string_view s) {
+    auto year = parse_date_time_fragment<std::chrono::year, int, '-'>(s);
```

Comment:
I would say the parsing in general looks very reasonable and I would have probably done it this way as well.

Maybe for the tests try add some weird manually constructed edge cases. I wouldn't expect anything to come up but never hurts to at least try to break it.

Another Idea would be to grep all the triples with date times out of swdf or something, try to parse it using `IStreamQuadIterator` and see if something comes up.